pkgs/by-name/ti/tinyfetch/package.nix

```diff
@@ -0,0 +1,32 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+}:
+
+stdenv.mkDerivation {
+  pname = "tinyfetch";
+  version = "0.2";
+
+  src = fetchFromGitHub {
+    owner = "abrik1";
+    repo = "tinyfetch";
+    rev = "c23c22d8af6cf60f15e18c9859623e2faff03dbc";
+    hash = "sha256-nuC7Xtfg7rf6/IDhUKT2JZD9V0Hl/welNAFDJYJKwmA=";
+  };
+
+  buildPhase = ''
```

Comment:
Remove this and change to:

```
  sourceRoot = "${src.name}/src";

  buildPhase = ''
    runHook preBuild
    $CC tinyfetch.c -o tinyfetch
    runHook postBuild
  '';

  installPhase = ''
    runHook preInstall
    install -Dm755 tinyfetch -t $out/bin
    runHook postInstall
  '';
```

Reply:
Wasn't able to get it to build with `sourceRoot = "${src.name}/src";`, was throwing this:
```
       error: undefined variable 'src'
       at /home/pagedmov/Nix/nixpkgs/pkgs/by-name/ti/tinyfetch/package.nix:18:19:
           17|
           18|   sourceRoot = "${src.name}/src";
             |                   ^
           19|
```

It builds fine like this, however:
```
  buildPhase = ''
    runHook preBuild
    ls
    $CC src/tinyfetch.c -o tinyfetch
    runHook postBuild
  '';

  installPhase = ''
    runHook preInstall
    install -Dm755 tinyfetch -t $out/bin
    runHook postInstall
  '';
```

I also changed the revision to use the 0.2 tag rather than the most recent commit.

Reply:
Update: added the missing rec in the function call to make ${src.name} work so the suggested changes have been integrated,